File: ios/MapErrorHandler.swift
import Foundation

private let mapsLogTag = "react-native-google-maps-plus"

final class MapErrorHandler {

  var callback: ((RNMapErrorCode, String) -> Void)?

  func report(_ code: RNMapErrorCode, _ msg: String, _ error: Error? = nil) {
    if let error {
      NSLog("[%@] %@ | %@", mapsLogTag, msg, String(describing: error))
    } else {
      NSLog("[%@] %@", mapsLogTag, msg)
    }

    onMain { [weak self] in
      guard let self else { return }
      self.callback?(code, msg)
    }
  }
}

File: android/src/main/java/com/rngooglemapsplus/MapErrorHandler.kt
package com.rngooglemapsplus

private const val MAPS_LOG_TAG = "react-native-google-maps-plus"

class MapErrorHandler {
  @Volatile
  var callback: ((RNMapErrorCode, String) -> Unit)? = null

  fun report(
    code: RNMapErrorCode,
    msg: String,
    t: Throwable? = null,
  ) {
    if (t != null) {
      android.util.Log.w(MAPS_LOG_TAG, msg, t)
    } else {
      android.util.Log.w(MAPS_LOG_TAG, msg)
    }

    onUi {
      callback?.invoke(code, msg)
    }
  }
}
